### DISCUSSION TITLE
Display three performances: capital, dividend and currency gains

### DISCUSSION THREAD

**Discussion by dtslvr** [ghostfolio/ghostfolio#1166] [Ideas]
Implementing the three components of performance would, in my view, greatly enhance the Ghostfolio experience.

To provide a dummy example, consider the purchase of a US short term bond ETF (denominated in USD) in the period Dec 2021-Jan 2022 by a user whose base currency is CHF. The performance currently calculated by Ghostfolio would be significantly negative (short-term bond ETFs lost between 3 and 6% since then). However, the currency effect entirely compensates for this loss (the USD has weakened >6% over the same period), i.e. the performance is actually positive when accounting for this factor. In addition, any distribution does not currently contribute to the performance calculation.

Hence, as I said before, it would be really useful to compute three different performance contributors:

- capital gains
- dividend gains
- currency gains

Let me know if I can support you, in case.
Thanks a lot!

_Originally posted by @FedericoComoglio in https://github.com/ghostfolio/ghostfolio/issues/554#issuecomment-1134898796_

**Comment by hvddrift** [comment]
This is a great addition. As I posted I uses excel with XIRR so these details would be great. The exchange rate addition (which I don’t do) is an excellent idea. 

Love to see this in a future version

**Comment by Stuart123456789** [comment]
Has there been any progress on this idea? Ghostfolio is an excellent implementation of a tracking tool and the addition of performance tracking across stock growth, dividend and currency effects would put it (even further) ahead of other portfolio tracking tools.

Let me know how I can help progress this idea further.

**Reply by dtslvr** [reply]
Hi @Stuart123456789

Thank you for your feedback and expressing your interest in enhancing Ghostfolio.

This idea has not yet been further developed. What would be your plan to break down the implementation into smaller tasks? Maybe start with the dividend gain?

**Reply by Stuart123456789** [reply]
@dtslvr - can you guide me to which file contains the current portfolio and individual holding value calculations? I'd like to understand better how the performance is currently calculated so I can better comment on how I would implement a change to this.
For example, I imagine different users have a preference for how the total value is displayed and it might be necessary to have a user changeable setting for the different performance calc modes.

**Reply by dtslvr** [reply]
Here is the entrypoint for the performance calculation: https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/portfolio/portfolio.controller.ts#L311

In a first step, it would be enough to additionally return the calculations for dividend and currency. Then present all of them separately in the user interface. In a last step, it would be cool to additionally have a combination of all three (overall performance). What do you think?

**Comment by Stuart123456789** [comment]
I'm making slow progress in understanding how to implement this. [edit: I did discover that the database pulls in historical exchange rate data. I think it hadn't fully updated yet]

The easiest place to start is in incorporating the dividend into the performance calculation. The API already calculates the dividend for each position. Given my slow progress, I'd like to start with just adding a column on the [Holdings table](https://github.com/ghostfolio/ghostfolio/blob/main/libs/ui/src/lib/holdings-table/holdings-table.component.html).

Can you help me understand if any of the datasources in the [.component.ts file](https://github.com/ghostfolio/ghostfolio/blob/main/libs/ui/src/lib/holdings-table/holdings-table.component.ts) can access the dividend for each position?

**Reply by dtslvr** [reply]
Yes, I agree to start with a more manageable task to get in. How about extending the endpoint `api/v1/portfolio/position/:dataSource/:symbol` by [_Dividend Yield_](https://www.investopedia.com/terms/d/dividendyield.asp)?

**Reply by Stuart123456789** [reply]
The API has a [getDividends](https://github.com/ghostfolio/ghostfolio/blob/b0fb986208ca29ca19a3abab8c49f87919e11c22/apps/api/src/app/portfolio/portfolio.controller.ts#L203C3-L203C29) section already. Would it be possible to use this to display the dividends per holding? The Dividend Yield becomes the calculation of `dividend value` / `value invested`.

**Comment by FedericoComoglio** [comment]
Hi everyone, just a quick comment to say that as a Ghostfolio user, I'm very pleased to see you got started on implementing this feature. Without a clear dissection of performance contributions, I am currently mainly using the app for logging activities and distributions.

**Comment by winklevos** [comment]
Another thing to add on this, consideration for interest. I don't have any suggestions on how to handle this in the overall performance. But given lending when the cash rate is high is a valid investment option it might need to be included in the overall.

**Comment by matzingerhh** [comment]
Hi I don't know that this is the right place but. So far I can see it is not possible to see dividends by year, right?

**Reply by dtslvr** [reply]
Annualized dividend yield is currently work in progress: https://github.com/ghostfolio/ghostfolio/issues/6238
